src/setup_payload/tests/run_python_setup_payload_test.py

```diff
@@ -62,6 +62,27 @@ def parse_setup_payload(chip_tool, payload):
     return parsed_params
 
 
+def chip_tool_generate_code(chip_tool, payload_data, cmd, key, is_qrcode):
```

Comment:
This is not OK --> this should be logic since it's not portable to call chip-tool

Reply:
I have added some test data set and parsing the payload and checking against it.

Keeping the earlier test for validating generated payload and parsing it with chip-tool, as suggested by @bzbarsky-apple https://github.com/project-chip/connectedhomeip/pull/24228#pullrequestreview-1235034454